parsec/core/gui/trio_jobs.py

```diff
@@ -54,48 +69,55 @@ def __init__(
         self._fn = fn
         self._args = args
         self._kwargs = kwargs
-        self.cancel_scope = None
+        self.cancel_scope: None | trio.CancelScope = None
         self._started = trio.Event()
         self._done = threading.Event()
-        self.status = None
-        self.ret = None
-        self.exc = None
+        self.status: None | str = None
+        self.ret: None | R = None
+        self.exc: None | trio.Cancelled | Exception | BaseExceptionGroup[
+            trio.Cancelled
+        ] | JobSchedulerNotAvailable = None
 
-    def __str__(self):
+    def __str__(self) -> str:
         return f"{self._fn.__name__}"
 
     @property
-    def arguments(self):
+    def arguments(self) -> OrderedDict[str, object]:
         bound_arguments = signature(self._fn).bind(*self._args, **self._kwargs)
         bound_arguments.apply_defaults()
         return bound_arguments.arguments
 
-    def is_finished(self):
+    def is_finished(self) -> bool:
         return self._done.is_set()
 
-    def is_cancelled(self):
+    def is_cancelled(self) -> bool:
         return self.status == "cancelled"
 
-    def is_ok(self):
+    def is_ok(self) -> bool:
         return self.status == "ok"
 
-    def is_errored(self):
+    def is_errored(self) -> bool:
         return self.status == "ko"
 
-    def is_crashed(self):
+    def is_crashed(self) -> bool:
         return self.status == "crashed"
 
-    async def __call__(self, *, task_status=trio.TASK_STATUS_IGNORED):
+    async def __call__(
+        self,
+        *,
+        # Mypy: `trio.TaskStatus` is provided by `trio_stubs`
+        task_status: trio.TaskStatus = trio.TASK_STATUS_IGNORED,  # type: ignore[name-defined]
```

Comment:
Why `ignore` needed, and what's the difference with `trio_typing.TaskStatus` ?

Reply:
I've redo the typing on that part, should be better now